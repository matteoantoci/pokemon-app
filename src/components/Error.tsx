import React from 'react'
import Typography from '@material-ui/core/Typography'
import ErrorIcon from '@material-ui/icons/Error'
import styled from 'styled-components'
import { red } from '@material-ui/core/colors'
import { Flex } from './Flex'

const Wrapper = styled(Flex)`
  flex-direction: column;
`

const IconWrapper = styled(Flex)`
  color: ${red[500]};
  justify-content: center;
  margin-bottom: 16px;
`

export const Error: React.FC = props => {
  return (
    <Wrapper {...props}>
      <IconWrapper>
        <ErrorIcon fontSize="large" />
      </IconWrapper>
      <Typography variant="h4" component="h5" align="center">
        Error while loading contents =(
      </Typography>
    </Wrapper>
  )
}
