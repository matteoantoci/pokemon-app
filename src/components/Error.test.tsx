import React from 'react'
import renderer from 'react-test-renderer'
import { Error } from './Error'

describe('Error', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Error />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
