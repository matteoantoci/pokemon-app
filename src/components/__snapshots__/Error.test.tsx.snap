// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correctly 1`] = `
<div
  className="sc-bwzfXH dhWCcq sc-bdVaJa kZAgjJ"
>
  <div
    className="sc-htpNat intnKj sc-bdVaJa kZAgjJ"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
      />
    </svg>
  </div>
  <h5
    className="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter"
  >
    Error while loading contents =(
  </h5>
</div>
`;
